riverdriver/riverpgxv5/migration/main/005_river_migration_add_line.up.sql

```diff
@@ -15,4 +15,35 @@ INSERT INTO river_migration
 SELECT created_at, 'main', version
 FROM river_migration_old;
 
-DROP TABLE river_migration_old;
\ No newline at end of file
+DROP TABLE river_migration_old;
+
+--
+-- Create `river_client` and derivative.
+--
+-- This feature hasn't quite yet been implemented, but we're taking advantage of
+-- the migration to add the schema early so that we can add it later without an
+-- additional migration.
+--
+
+CREATE UNLOGGED TABLE river_client (
+    name text PRIMARY KEY NOT NULL,
```

Comment:
notably this creates a hard constraint on unique client names, which may be breaking when rolled out

Reply:
Hmm, so default names should always produce relatively unique names. You're thinking if someone overrides `ID` and names everything the same thing?

I think probably what would happen in that case rather than throwing an error would be that the stats produced would end up being nonsense since the clients would be overriding each other's work. Not great, but at least it's not too harmful beyond bad numbers.

Reply:
Yeah, specifically for overriding. Like what if somebody used Heroku style `web.1` names instead of the unique dyno ID. That may interfere with leader election anyway. Just wanted to be mindful of this as we implement it, it's probably fine for a stats table as long as it doesn't become an extra breaking point.

Reply:
I guess we could do something like try to detect that a newly started client updates a record that was inserted recently instead of inserting a fresh one, which might suggest a duplicate ID, but even that could only be a warning at best because it could be legitimate restart that just happened to have occurred quite quickly since the last one.

But yeah, I'm not too compelled to fix a problem that'd be entirely user-induced. There's only so much you can do to protect people from themselves, and like you point out, leader election won't work either.